src/language/ast.js

```diff
@@ -44,6 +44,30 @@ export type Location = {
 };
 
 /**
+ * Represents the different kinds of tokens in a GraphQL document.
+ */
+export type TokenKind = '<SOF>'
```

Comment:
Didn’t know that existed. At a minimum we shouldn’t export `TokenKind` then.

Does Flow have a way to get the type of all the values? I know that TypeScript has `keyof` which will union all the string literal key types. Something like [`$PropertyType<T, x>`](https://flowtype.org/docs/utility-types.html#propertytypet-x), but it returns a union of _all_ the property types and not just one?

Assuming the property types are all string literals we could then write:

```js
type Token = {
  kind: $PropertyTypes<TokenKind>,
  ...
};
```

This would also be a reverse [`$Keys<T>`](https://flowtype.org/docs/utility-types.html#keyst).